src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1080,7 +1080,9 @@ private void emitStaticOverloads(String propName, ObjectType objType, JSType pro
       }
       // "I, for one, welcome our new static overloads." - H.G. Wells.
       JSType superPropType = superTypeCtor.getPropertyType(propName);
-      if (!superPropType.equals(propertyType)) {
+      if (!superPropType.isSubtype(propertyType)) {
+        // If the super field is private we have no issue.
```

Comment:
Maybe add `because private fields do not get emitted`? TypeScript itself would bail.